Tests/Factory/ResponseExceptionFactoryTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * This file is part of DoctrineRestDriver.
+ *
+ * DoctrineRestDriver is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * DoctrineRestDriver is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DoctrineRestDriver.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Circle\DoctrineRestDriver\Tests\Factory;
+
+use Circle\DoctrineRestDriver\Factory\ResponseExceptionFactory;
+use Symfony\Component\HttpFoundation\Response;
+use Doctrine\DBAL\Driver\DriverException as DriverExceptionInterface;
+use Doctrine\DBAL\Exception as DBALException;
+
+/**
+ * @coversDefaultClass Circle\DoctrineRestDriver\Factory\ResponseExceptionFactory
+ * 
+ * @author Rob Treacy <robert.treacy@thesalegroup.co.uk>
+ */
+class ResponseExceptionFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    private $responseExceptionFactory;
+    
+    public function setUp() {
+        $this->responseExceptionFactory = new ResponseExceptionFactory;
+    }
+    
+    /**
+     * @test
+     * @group  unit
+     * @covers ::createDbalException
+     * @dataProvider createDbalExceptionProvider
+     */
+    public function createDbalException(Response $response, $expectedExceptionClass) {
+        $return = $this->responseExceptionFactory->createDbalException($response, $this->createMock(DriverExceptionInterface::class));
+        $this->assertInstanceOf($expectedExceptionClass, $return);
+    }
+    
+    public function createDbalExceptionProvider() {
```

Comment:
Data providers for unit tests need to be public so that PHPUnit can access it to get each set of arguments for the test method itself.